packages/core/src/routes/connector.ts

```diff
@@ -208,4 +208,39 @@ export default function connectorRoutes<T extends AuthedRouter>(router: T) {
       return next();
     }
   );
+
+  router.delete(
+    '/connectors/:id',
+    koaGuard({ params: object({ id: string().min(1) }) }),
+    async (ctx, next) => {
+      const {
+        params: { id },
+      } = ctx.guard;
+
+      const databaseEntries = await findAllConnectors();
+      const databaseEntry = databaseEntries.find((entry) => entry.id === id);
+
+      if (!databaseEntry) {
+        throw new RequestError({
+          code: 'entity.not_found',
+          id,
+          status: 422,
+        });
+      }
+
+      const virtualConnectors = await loadConnectors();
+      const virtualConnector = virtualConnectors.find(
+        ({ metadata: { id } }) => id === databaseEntry.connectorId
+      );
+
+      if (virtualConnector) {
+        assertThat(!databaseEntry.enabled, 'connector.delete_enabled_connector_failed');
```

Comment:
How can the user set a connector's enabled to false?

Reply:
Users should go to the "Connector Details Page" to disable connectors and fix SIE before deleting a connector instance.
I think the guide should be given by AC after catching this error.

Reply:
Removed this check.